ouroboros-consensus/ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Protocol/Praos.hs

```
@@ -128,25 +129,22 @@ deriving instance PraosCrypto c => Show (HotKey c)
 instance PraosCrypto c => NoUnexpectedThunks (HotKey c) where
   showTypeOf _ = show $ typeRep (Proxy @(HotKey c))
 
-evolveKey :: (Monad m, PraosCrypto c)
-          => SlotNo -> HotKey c -> m (HotKey c)
-evolveKey slotNo (HotKey oldKey) = do
-    let newKey = fromMaybe (error "evolveKey: updateKES failed") $
-                 updateKES () oldKey kesPeriod
-    return $ HotKey newKey
+evolveKey :: PraosCrypto c => SlotNo -> HotKey c -> HotKey c
+evolveKey slotNo (HotKey oldKey) =
+    HotKey $ fromMaybe (error "evolveKey: updateKES failed") $
+               updateKES () oldKey kesPeriod
```

When `updateKES` returns `Nothing`, let the update fail.